jfan200/ReactStudyNote
# Task/authentication and backend

## Describe your changes
- Set up the initial backend structure for the project.
- Added models for User, Building, Task, Job, Feedback, and Image.
- Configured MongoDB connection using Bson serialization attributes.
- Included essential project configuration files (e.g., appsettings and launch settings).
- Implemented repository pattern with Repo and IRepo for data access.
- Added AuthHandler for managing authentication logic.
- Set up ApiController to handle API requests and responses.
- Added Data Transfer Objects (Dto) to structure data communication between layers.

## Issue ticket number and link
- [CP-117](https://comp90082-teamcp.atlassian.net/browse/CP-117?atlOrigin=eyJpIjoiMzQwMTU4NTRlNGU0NDQyNmE5NTMzMTZmMmIwY2EzZDYiLCJwIjoiaiJ9)
- [CP-206](https://comp90082-teamcp.atlassian.net/browse/CP-206?atlOrigin=eyJpIjoiYjE2N2I4YjkwNzQwNDNkMDg1ZjlkZDFjNTE2ZTc1NWYiLCJwIjoiaiJ9)
- [CP-126](https://comp90082-teamcp.atlassian.net/browse/CP-126?atlOrigin=eyJpIjoiMTcxOGJiMzA2ZDI1NDBmNmEzN2U0MmUxMDI1NTk5MDAiLCJwIjoiaiJ9)

## Self Check before Merge
- [x] I have added/updated tests if this is a core feature.
- [ ] I have communicated this PR review to mandatory approvers.
- [ ] I have updated documentation on README.
- [ ] I have updated documentation on Confluence.

## How can a reviewer test this code?
	1.	Clone the repository and switch to the relevant branch.
	2.	Ensure MongoDB is running and the necessary connection strings are correctly set in the appsettings.json file.
	3.	Run the backend project locally using Visual Studio or through the CLI (dotnet run).
	4.	Use Postman or SwaggerUI to send API requests to the available endpoints provided by the ApiController.
	5.	Verify that the CRUD operations for User, Building, Task, Job, Feedback, and Image models work as expected.
	6.	Confirm the authentication logic through the AuthHandler by testing protected routes.
	7.	Review the data transfer process via the Dto by inspecting the response formats.

## How to deploy this code?
	1.	Pull the latest code from the repository.
	2.	Ensure MongoDB is correctly configured in the production environment.
	3.	Update the necessary environment variables, such as connection strings in the appsettings.Production.json file.
	4.	Build and deploy the backend project using the dotnet publish command, or through your preferred CI/CD pipeline.
	5.	Deploy the API server to your hosting environment (e.g., AWS, Azure, etc.).
	6.	Verify the deployment by sending requests to the live API and confirming functionality.

## Useful Links
- [Postman API documentation: ](https://documenter.getpostman.com/view/27616614/2sAXqqcNkF)
